lib/src/components/NxTextInput/NxTextInput.tsx

```diff
@@ -96,7 +94,7 @@ const NxTextInput = forwardRef<TextInputElement, Props>(
 
       function inputOnChange(e: FormEvent<TextInputElement>) {
         if (onChange) {
-          onChange(e.currentTarget.value);
+          onChange(e.currentTarget.value, e);
```

Comment:
Personally, I think these should be flipped so that we can use the `onChange` just like the normal event from React. That would make this a breaking change though, so I guess this is fine.

Reply:
I can definitely see that argument, but seems like keeping backwards compatibility trumps that, plus I buy Ross's idea that in many cases it would be better to use the abstracted value